packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/base/io.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+final String flutterRootPath = getFlutterRoot();
+final Directory flutterRoot = fileSystem.directory(flutterRootPath);
+
+Future<void> main() async {
+  // Regression test for https://github.com/flutter/flutter/issues/132592
+  test('flutter/bin/dart updates the Dart SDK', () async {
+    // Run the Dart entrypoint once to ensure the Dart SDK is downloaded.
+    await runDartBatch();
+
+    // Remove the Dart SDK stamp to cause the Dart batch entrypoint to
+    // re-download the Dart SDK.
+    expect(dartSdkStamp.existsSync(), true);
+    dartSdkStamp.deleteSync();
+
+    // Run the Dart batch entrypoint again to trigger the Dart SDK update.
+    final stdout = await runDartBatch();
+
+    // Check the Dart SDK was re-downloaded and extracted without overwriting
+    // unexpected files.
+    // See: https://github.com/flutter/flutter/issues/132592
+    expect(stdout, contains('Downloading Dart SDK from Flutter engine ...'));
+    expect(stdout, contains('Expanding downloaded archive...'));
+    expect(stdout, isNot(contains('Use the -Force parameter')));
```

Comment:
So when I run this test locally without the powershell change from this PR, it hangs indefinitely (as expected). @Hixie is this a reasonable place to set a timeout with an explicit `fail()`, that way if it starts failing, people don't just spam re-run?

Reply:
in discord, Hixie had a good idea, to set a Timer for something less than the overall test runner timeout time, which I believe is 12 minutes, and have that timer print out a message saying:

> this is running for much longer than expected and this test is explicitly verifying something doesn't hang, so if it times out that means the test has failed and is unlikely to be a flake.

Reply:
Added! :)